pkg/storage/tsdb/block/index.go

```diff
@@ -620,22 +621,22 @@ type seriesRepair struct {
 // which index.Reader does not implement.
 type indexReader interface {
 	Symbols() index.StringIter
-	SortedLabelValues(name string, matchers ...*labels.Matcher) ([]string, error)
-	LabelValues(name string, matchers ...*labels.Matcher) ([]string, error)
-	Postings(name string, values ...string) (index.Postings, error)
+	SortedLabelValues(ctx context.Context, name string, matchers ...*labels.Matcher) ([]string, error)
+	LabelValues(ctx context.Context, name string, matchers ...*labels.Matcher) ([]string, error)
+	Postings(ctx context.Context, name string, values ...string) (index.Postings, error)
 	SortedPostings(index.Postings) index.Postings
```

Comment:
little bit weird that these 3 don't have context, but I guess these are not on the hotpath/weren't modified in Prometheus either

Reply:
Correct, they weren't modified in Prometheus. Maybe we should review at a later point if there's a need.